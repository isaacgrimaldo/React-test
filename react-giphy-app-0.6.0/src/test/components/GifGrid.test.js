import React from 'react';
import'@testing-library/jest-dom';
import {shallow} from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import {useFetchGifs} from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs') // Para controlar la informacion que le estamos mandando

describe('Puebas de archivo GifGrid', () => {
    const category ='Goku';
    
    test('Bebe de mostrar se correctamente el componente', () => {
         
        useFetchGifs.mockReturnValue({ 
            data:[],
            loading: true
        });

        const wrapper  =  shallow( <GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de moratrar la informacion que viene de useFetchGifs', () => {
        
        const gifs = [
            {
                id:'ABC',
                url:'http://localhost/lago.png',
                title:'Algo papa'
            }
         ]
        useFetchGifs.mockReturnValue({ 
            data:gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category ={ category }/> );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })
})