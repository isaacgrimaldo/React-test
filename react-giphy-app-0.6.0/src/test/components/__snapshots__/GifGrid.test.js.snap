// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas de archivo GifGrid Bebe de mostrar se correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Puebas de archivo GifGrid Debe de moratrar la informacion que viene de useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Algo papa"
      url="http://localhost/lago.png"
    />
  </div>
</Fragment>
`;
